core/base-service/auth-helper.js

```diff
@@ -0,0 +1,43 @@
+'use strict'
+
+class AuthHelper {
+  constructor({ userKey, passKey, isRequired = false }, privateConfig) {
+    if (!userKey && !passKey) {
+      throw Error('Expected userKey or passKey to be set')
+    }
+
+    this._userKey = userKey
+    this._passKey = passKey
+    this.user = userKey ? privateConfig[userKey] : undefined
+    this.pass = passKey ? privateConfig[passKey] : undefined
+    this.isRequired = isRequired
+  }
+
+  get isConfigured() {
+    return (
+      (this._userKey ? Boolean(this.user) : true) &&
+      (this._passKey ? Boolean(this.pass) : true)
+    )
+  }
+
+  get isValid() {
```

Comment:
Is this going to be used when the service classes (the ones using auth) are instantiated or when requests are made? I _believe_ it's the latter if I'm following the PR summary and some of the below files correctly

Reply:
The use case I originally had in the back of my mind (trying to catch config errors as early as possible) may not be as applicable. However, I'm not sure that's all that relevant. For example, SymfonyInsight service requires authorization, but I don't set that in my self-hosted Shields deployment (we don't need it) and I wouldn't want to _have_ to set it either. 

IIRC we have a similar setup in the PR review/staging apps and CI jobs where services like Wheelmap and Symfony don't have the required auth configured either. It might be nice to catch an invalid config at deploy time for a Product Shields.io deployment though (if we don't already have something like that).

Reply:
> Is this going to be used when the service classes (the ones using auth) are instantiated or when requests are made? I _believe_ it's the latter if I'm following the PR summary and some of the below files correctly

For now it's when requests are made, though I agree it would be better to examine the config at load time. That would mean moving it to `register()`, where I had it initially, but it made testing really annoying because you couldn't call `invoke()` without running the auth config first.

It would be nice to bail on startup if services are _partially_ configured (i.e. failing as loudly as possible).

I'd be open to improving/revisiting this after #3328.